st2common/st2common/hooks.py

```diff
@@ -37,14 +37,38 @@ def after(self, state):
 
         origin = state.request.headers.get('Origin')
         origins = cfg.CONF.api.allow_origin
+        serve_webui_files = cfg.CONF.api.serve_webui_files
+
+        # Build a list of the default allowed origins
+        api_port = cfg.CONF.api.port
+        public_api_url = cfg.CONF.auth.api_url
+        default_allowed_origins = []
+
+        # Default WebUI URL
+        default_allowed_origins.append('http://localhost:3000')
+
+        if serve_webui_files:
+            # Local API URL
+            default_allowed_origins.append('http://localhost:%s' % (api_port))
+            default_allowed_origins.append('http://127.0.0.1:%s' % (api_port))
+
+        if serve_webui_files and public_api_url:
+            # Public API URL
+            default_allowed_origins.append(public_api_url)
+
```

Comment:
On one hand, it might be confusing for someone who knows how it works. On the other hand, it might same some time for those who don't. I don't see any security risk here, so I say we should keep and carefully document that. Like, "we are adding origins A,B and C for your convenience, but we still suggest you to add every origin you are going to use manually".